src/Jackett/Indexers/WiHD.cs

```diff
@@ -0,0 +1,702 @@
+
+            // Test if we are logged in
+            await ConfigureIfOK(response.Cookies, response.Content != null && response.Content.Contains("/logout"), () => {
+                // Oops, unable to login
+                throw new ExceptionWithConfigData("Failed to login", configData);
+            });
+
+            return IndexerConfigurationStatus.RequiresTesting;
+        }
+
+        /// <summary>
+        /// Execute our search query
+        /// </summary>
+        /// <param name="query">Query</param>
+        /// <returns>Releases</returns>
+        public async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var torrentRowList = new List<CQ>();
+            var searchTerm = query.GetQueryString();
+            var searchUrl = SearchUrl;
+
+            // Check cache first so we don't query the server
+            if(!DevMode)
+            {
+                lock (cache)
+                {
+                    // Remove old cache items
+                    CleanCache();
+
+                    var cachedResult = cache.Where(i => i.Query == searchTerm).FirstOrDefault();
+                    if (cachedResult != null)
+                        return cachedResult.Results.Select(s => (ReleaseInfo)s.Clone()).ToArray();
+                }
+            }
+
+            // Add emulated XHR request
+            emulatedBrowserHeaders.Add("X-Requested-With", "XMLHttpRequest");
+
+            // Request our first page
+            latencyNow();
+            var results = await RequestStringWithCookiesAndRetry(buildQuery(searchTerm, query, searchUrl), null, null, emulatedBrowserHeaders);
+            fDom = results.Content;
+
+            try
+            {
+                // Find number of results
+                int nbResults = ParseUtil.CoerceInt(Regex.Match(fDom["div.ajaxtotaltorrentcount"].Text(), @"\d+").Value);
+                output("\nFound " + nbResults + " results for query !");
+
+                // Find torrent rows
+                var firstPageRows = findTorrentRows();
+                output("There are " + firstPageRows.Length + " results on the first page !");
+                torrentRowList.AddRange(firstPageRows.Select(fRow => fRow.Cq()));
+
+                // Calculate numbers of pages available for this search query
+                int pageLinkCount = (int)Math.Ceiling((double)nbResults / firstPageRows.Length);    // Based on number results and number of torrents on first page
+                output("--> Pages available for query: " + pageLinkCount);
+
+                // If we have a term used for search and pagination result superior to one
+                if (!string.IsNullOrWhiteSpace(query.GetQueryString()) && pageLinkCount > 1)
+                {
+                    // Starting with page #2
+                    for (int i = 2; i <= Math.Min(Int32.Parse(ConfigData.Pages.Value), pageLinkCount); i++)
```

Comment:
Software does make automated requests for backfills which may hit the server lots.  Limit this to three calls.